src/caip-types.ts

```diff
@@ -46,6 +46,12 @@ export const CaipAccountAddressStruct = pattern(
 );
 export type CaipAccountAddress = Infer<typeof CaipAccountAddressStruct>;
 
+/** Supported CAIP namespaces. */
+export const CaipNamespaces = {
+  /** EIP-155 compatible chains. */
+  Eip155: 'eip155' as CaipNamespace,
```

Comment:
Why do we need to use a type assertion? Would it make more sense to use an `enum` here?

Reply:
Indeed `enum` makes more sense here, I wonder if I had a use-case for it to be a `const` object... Lemme look at this quickly. :)

Reply:
IIRC I wanted to force it to "really be" of type `CaipNamespace` (which is a `string right now), in case this would change in the future but... If that type changes, we would also have to update the literal `'eip155'` too...

Lemme use an `enum` here, it's definitely better in this context.

Reply:
Taking a closer look at this, it's customary in TypeScript to name enums using singular rather than plural. This is because each of the options in the enum can be regarded as satisfying the same type. For instance, if you were to write a function that took the enum, it would make sense to say that it took a `CaipNamespace` rather than a `CaipNamespaces`.

However, if you named this enum `CaipNamespace`, it would create a conflict, because we already have a type called `CaipNamespace`. So... what does this enum represent vs. the existing `CaipNamespace`? Is it fair to say that the existing `CaipNamespace` represents a _theoretically valid_ CAIP namespace but your enum represents real, _known_ CAIP namespaces?

Should we rename the existing `CaipNamespace` struct and types so that we can give this enum a complementary name?

Also bringing @jiexi into this conversation since he was the one who added the original types :)

Reply:
Yes, that is exactly why I named it `CaipNamespaces` to avoid name conflict at first.

And yes that is correct, this enum currently holds known CAIP namespaces that we plan to use in our Snap keyring implementations.

Instead of having a literal like `'eip155'` everywhere in the codebase, I thought having a enum/constant value would be better in terms of readability.

Given the current structure of the file, I believe `CaipNamespace` is good enough for a type name. Maybe it's also time to have a `caip-constants.ts` and move it there. But even with that, we would probably need to change the enum name.

Reply:
Do types and enums/constants for the [network in extension](https://github.com/MetaMask/metamask-extension/blob/646a6af0316a08d68410c2487ab9002a060bee6a/shared/constants/network.ts#L6) give us any inspiration?

Reply:
so I could probably go with `CAIP_NAMESPACES` then and use `const` objects (if we follow the same pattern). Sounds ok to me. WDYT @mcmire?

Also, do we want a `caip.ts` for those constants? or a `caip-constants.ts` maybe?

Reply:
> Do types and enums/constants for the [network in extension](https://github.com/MetaMask/metamask-extension/blob/646a6af0316a08d68410c2487ab9002a060bee6a/shared/constants/network.ts#L6) give us any inspiration?

In this file, we pair a `SCREAMING_SNAKE_CASE` constant with a `PascalCase` type. I believe that this pattern is a carryover from when this file was written in JavaScript. But this is the same thing that `enum` does, and I feel like that's what we'd have used if we'd written this file from scratch. So I don't know that we need to copy this pattern; I feel that it would be simpler to directly use an `enum`.

What are our thoughts on renaming the existing type (the existing `CaipNamespace`) `ValidCaipNamespace` and this new enum `KnownCaipNamespace`? I think that would communicate the intended purpose of these types.

> do we want a caip.ts for those constants? or a caip-constants.ts maybe?

If the current name is confusing, I think renaming it to `caip.ts` makes sense, assuming that it doesn't mess up the diff.

Reply:
feels weird consuming `ValidCaipNamespace`. I think leaving `CaipNamespace` as is and adding a new `KnownCaipNamespace` is a good compromise

Reply:
I'm definitely ok with that. Let's keep `CaipNamespace` as is, and I'll rename my `enum` to `KnownCaipNamespace`. Thanks guys! :)

Reply:
@jiexi @mcmire I've updated the PR, lemme know if you want me to change anything else!